### ISSUE TITLE
`extend schema @link(...) { query: Query }` causes error in buildSubgraph

### ISSUE THREAD

**Issue by lennyburdette** [apollographql/federation#1875]
Failing test: https://github.com/apollographql/federation/compare/main...lennyburdette:extend-schema-parse-error?expand=1

A customer encountered this when their sdl merging script (based on `@graphql-tools/merge`) resulted in:
```graphql
extend schema
  @contact(...)
  @link(url: "https://specs.apollo.dev/federation/v2.0" import: ["@key", "@shareable", "@tag"]) {
  query: Query
  mutation: Mutation
}
```

The exception thrown is 
```
GraphQLError: Syntax Error: Expected "{", found Name "extend".
```

I printed out the intermediary SDL in `Schema.toAST` and got:

```graphql
schema
  @link(url: "https://specs.apollo.dev/link/v1.0")

extend schema
  @link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@tag"])
{
  query: Query
}

directive @link(url: String, as: String, for: link__Purpose, import: [link__Import]) repeatable on SCHEMA

directive @federation__key(fields: federation__FieldSet!, resolvable: Boolean = true) repeatable on OBJECT | INTERFACE

directive @federation__requires(fields: federation__FieldSet!) on FIELD_DEFINITION

directive @federation__provides(fields: federation__FieldSet!) on FIELD_DEFINITION

directive @federation__external(reason: String) on OBJECT | FIELD_DEFINITION

directive @tag(name: String!) repeatable on FIELD_DEFINITION | OBJECT | INTERFACE | UNION | ARGUMENT_DEFINITION | SCALAR | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION

directive @federation__extends on OBJECT | INTERFACE

directive @federation__shareable on OBJECT | FIELD_DEFINITION

directive @federation__inaccessible on FIELD_DEFINITION | OBJECT | INTERFACE | UNION | ARGUMENT_DEFINITION | SCALAR | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION

directive @federation__override(from: String!) on FIELD_DEFINITION

type Query {
  hello: String
  _service: _Service!
}

enum link__Purpose {
  """
  `SECURITY` features provide metadata necessary to securely resolve fields.
  """
  SECURITY

  """
  `EXECUTION` features provide metadata necessary for operation execution.
  """
  EXECUTION
}

scalar link__Import

scalar federation__FieldSet

scalar _Any

type _Service {
  sdl: String
}
```

**Comment by sachindshinde** [comment]
@lennyburdette 
It's worth noting that
```graphql
schema
  @link(url: "https://specs.apollo.dev/link/v1.0")
```
is invalid GraphQL according to spec. Specifically, the [spec grammar](https://spec.graphql.org/draft/#sec-Schema) says
> [_Description_](https://spec.graphql.org/draft/#Description)<sub>opt</sub> **schema** [_Directives_](https://spec.graphql.org/draft/#Directives)<sub>[Const]opt</sub> { [_RootOperationTypeDefinition_](https://spec.graphql.org/draft/#RootOperationTypeDefinition)<sub>list</sub> }

The gist is that the curly braces are required, and there must be at least one root operation type definition in the braces.

(Also technically [as per spec](https://spec.graphql.org/draft/#sec-Schema-Extension.Schema-Validation), `extend schema` definitions can't be used without an original `schema` definition in the schema, but `graphql-js` has never enforced this, so it's effectively become de-facto valid.)

**Comment by lennyburdette** [comment]
Yep that all makes sense! I’m not sure why federation-internal generates that invalid schema.

**Comment by pcmanus** [comment]
Opened fix as #1882.

The issue was that as we "expand" the subgraph definition, adding missing definitions and all, we add `@link(url: "https://specs.apollo.dev/link/v1.0")` and we were always doing it on the schema _definition_ (so `schema`, not `extend schema`). Now, if you follow the graphQL spec to the letter, then that's fine. But due to the leeway around having `extend schema` without `schema`, we can be in a situation like this one were we cannot properly print the schema if we try to put the directive on the schema definition.

Now, we're the first to abuse this leeway in our examples, using "naked" `extend schema` to save some typing, so I'm certainly not going to suggest against fixing this. And what the attached patch does is simply to put the `@link(url: "https://specs.apollo.dev/link/v1.0")` on the schema extension in this kind of case (which to be clear, is essentially invisible for the user, but means the code don't crash anymore).
